### PR TITLE
LEAR-15-Working-with-request-ButonClicked-without-State-like1_1(1_+)-…

### PR DESCRIPTION
**Pull request by Andstani**
…and-request-with-State.

### CODE REVIEW

File: Calculator.Logic/CalculatorService.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            _calculatorLogic = calculatorLogic;
+        }
+        public CalculatorResponse GetCalculatorResponse(CalculatorRequest request)
+        {
+            var calculatorResponse = new CalculatorResponse();
+            var strings = request.ButtonClicked.Split('_').ToList();
+
+            SetTheCalculatorStateIfExistsInRequest(request);
+            if (strings != null && strings.Count > 0)
```
Comment (AvramRotar) [inline Calculator.Logic/CalculatorService.cs]: Extract this if in another method

Comment (AvramRotar) [inline Calculator.Logic/CalculatorService.cs]: strings.Count should equal 2

File: Calculator.Logic/CalculatorService.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                _calculatorLogic.AddTerm(strings[0]);
+
+                if (strings[1] == "=")
```
Comment (AvramRotar) [inline Calculator.Logic/CalculatorService.cs]: extract this in another method (ProcesOperation ?)

File: Calculator.Logic/CalculatorService.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                            break;
+                        default:
+                            calculatorResponse.CalculatorState = _calculatorLogic.GetCalculatorState();
+                            break;
+                    }
+                }
+            }
+            return calculatorResponse;
+        }
+            private static Operation ParseStringToEnumOperation(List<string> strings)
```
Comment (AvramRotar) [inline Calculator.Logic/CalculatorService.cs]: this method should not exist

File: Calculator.Logic/CalculatorService.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                _calculatorLogic.AddTerm(strings[0]);
+
+                if (strings[1] == "=")
+                {
+                    calculatorResponse.CalculatorResult = _calculatorLogic.DoMultipleTermOperation();
+                }
+                else
+                {
+                    Operation operation = ParseStringToEnumOperation(strings);
```
Comment (AvramRotar) [inline Calculator.Logic/CalculatorService.cs]: Do single Term and Do multiple term operation should be different flows

File: Calculator.API/Program.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
```
Comment (AvramRotar) [inline Calculator.API/Program.cs:13]: builder.Services.AddControllers().AddJsonOptions(options =>
        {
            options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
        });
        add this to have string values mapping form enum

File: Calculator.Logic/CalculatorService.cs (hunk 1 above)
Comment (Andstani) [reply Calculator.Logic/CalculatorService.cs]: Done

**Comment by Andstani** [comment]
Am putea sa facem Merge? Cand ma schimb pe LEAR-16 imi lipseste codul din LEAR-15 si imi da erroare. Ma gandesc ca se rezolva dupa ce facem Merge la LEAR-15.

Multumesc

:D

**Pull request merged**